### ISSUE TITLE
Failed build - undefined reference to vtable from main.cpp

### ISSUE THREAD

**Issue by austin-millan** [w8rbt/oathgen#9]
**Background:**
OS: Ubuntu 18.04
libcrypto packages: 
- [`libcrypto++6`](https://packages.debian.org/sid/amd64/libcrypto++6), 
- `libcrypto++6-dbg`, 
- [`libcrypto++-dev`](https://packages.debian.org/sid/libcrypto++-dev)

**Building:**

When I first tried building I ran into the same issue as #4, which was an easy fix by updating the build script and following the instructions under `alt_base32/readme.txt`:

```
$ sudo cp -r alt_base32/* /usr/include/cryptopp/
```

When I tried to build again I got a more interesting error:
```
./build.sh linux
/tmp/ccrP2CpF.o: In function `CryptoPP::AltBase32Decoder::AltBase32Decoder(CryptoPP::BufferedTransformation*)':
main.cpp:(.text._ZN8CryptoPP16AltBase32DecoderC2EPNS_22BufferedTransformationE[_ZN8CryptoPP16AltBase32DecoderC5EPNS_22BufferedTransformationE]+0x16): undefined reference to `CryptoPP::AltBase32Decoder::GetDefaultDecodingLookupArray()'
main.cpp:(.text._ZN8CryptoPP16AltBase32DecoderC2EPNS_22BufferedTransformationE[_ZN8CryptoPP16AltBase32DecoderC5EPNS_22BufferedTransformationE]+0x34): undefined reference to `vtable for CryptoPP::AltBase32Decoder'
main.cpp:(.text._ZN8CryptoPP16AltBase32DecoderC2EPNS_22BufferedTransformationE[_ZN8CryptoPP16AltBase32DecoderC5EPNS_22BufferedTransformationE]+0x42): undefined reference to `vtable for CryptoPP::AltBase32Decoder'
/tmp/ccrP2CpF.o: In function `CryptoPP::AltBase32Decoder::~AltBase32Decoder()':
main.cpp:(.text._ZN8CryptoPP16AltBase32DecoderD2Ev[_ZN8CryptoPP16AltBase32DecoderD5Ev]+0xf): undefined reference to `vtable for CryptoPP::AltBase32Decoder'
main.cpp:(.text._ZN8CryptoPP16AltBase32DecoderD2Ev[_ZN8CryptoPP16AltBase32DecoderD5Ev]+0x1d): undefined reference to `vtable for CryptoPP::AltBase32Decoder'
collect2: error: ld returned 1 exit status
```

I tried compiling with a different standard/version (`-std=c++0x`) with no luck. Do you have any suggestions on getting this to work on Debian systems? Any help would be appreciated.

Thanks for your time!

**Comment by w8rbt** [comment]
Thanks for the note Austin. I've updated Oathgen. I've meant to do this for some time now. I've removed the alt_base32 files (which was a workaround for a bug in base32 decoding) and now, I use the built-in Crypto++ base32 library (with an alternate alphabet). I built and tested Oathgen with Crypto++ 8.0 release and it seems to work fine on Debian and Ubuntu. Hope this helps, and thanks again. Brad

**Comment by w8rbt** [comment]
See this thread for more info on the old bug in Crypto++ base32 decoding:

    https://github.com/weidai11/cryptopp/issues/108

I think Crypto++ 5.6.4 and beyond fixed it, but I only verified the fix with the Crypto++ 8.0 and Oathgen 1.1 releases.

**Comment by austin-millan** [comment]
Thanks for getting back to me, indeed it seems to be a bug in the Crypto++ version.
